de.metas.business/src/main/java/org/adempiere/impexp/DiscountSchemaImportProcess.java

```diff
@@ -0,0 +1,224 @@
+			InterfaceWrapperHelper.save(bpartner);
+		}
+		else
+		{
+			discountSchema = importRecord.getM_DiscountSchema();
+		}
+
+		ModelValidationEngine.get().fireImportValidate(this, importRecord, discountSchema, IImportInterceptor.TIMING_AFTER_IMPORT);
+		InterfaceWrapperHelper.save(discountSchema);
+		importRecord.setM_DiscountSchema_ID(discountSchema.getM_DiscountSchema_ID());
+		return inventoryImportResult;
+	}
+
+	private I_M_DiscountSchema createNewMDiscountSchemas(I_I_DiscountSchema importRecord)
+	{
+		final I_M_DiscountSchema schema;
+		schema = InterfaceWrapperHelper.create(getCtx(), I_M_DiscountSchema.class, ITrx.TRXNAME_ThreadInherited);
+		schema.setAD_Org_ID(importRecord.getAD_Org_ID());
+		schema.setValidFrom(SystemTime.asDayTimestamp());
+		schema.setDiscountType(X_M_DiscountSchema.DISCOUNTTYPE_Breaks);
+		schema.setName("I " + importRecord.getM_DiscountSchema_ID());
+		schema.setDescription("I " + importRecord.getC_BPartner().getValue() + " " + importRecord.getM_Product().getValue());
+		return schema;
+	}
+
+	/**
+	 * reuse previous discount schema
+	 *
+	 * @param importRecord
+	 * @param previousImportRecord
+	 * @return
+	 */
+	private ImportRecordResult doNothingAndUsePreviousDiscountSchema(@NonNull final I_I_DiscountSchema importRecord, @NonNull final I_I_DiscountSchema previousImportRecord)
+	{
+		importRecord.setM_DiscountSchema_ID(previousImportRecord.getM_DiscountSchema_ID());
+		return ImportRecordResult.Nothing;
+	}
+
+	private I_M_DiscountSchemaBreak importDiscountSchemaBreak(@NonNull final I_I_DiscountSchema importRecord)
+	{
+		final I_M_DiscountSchema schema = importRecord.getM_DiscountSchema();
+
+		I_M_DiscountSchemaBreak schemaBreak = importRecord.getM_DiscountSchemaBreak();
+		if (schemaBreak != null)
+		{
+			if (schemaBreak.getM_DiscountSchema_ID() <= 0)
+			{
+				schemaBreak.setM_DiscountSchema(schema);
+			}
+			else if (schemaBreak.getM_DiscountSchema_ID() != schema.getM_DiscountSchema_ID())
+			{
+				throw new AdempiereException("Discount schema of Discount schema break <> Discount schema");
+			}
+
+			setDiscountSchemaBreakFields(importRecord, schemaBreak);
+
+			ModelValidationEngine.get().fireImportValidate(this, importRecord, schemaBreak, IImportInterceptor.TIMING_AFTER_IMPORT);
+			InterfaceWrapperHelper.save(schemaBreak);
+		}
+		else
+		{
+			schemaBreak = InterfaceWrapperHelper.create(getCtx(), I_M_DiscountSchemaBreak.class, ITrx.TRXNAME_ThreadInherited);
+			setDiscountSchemaBreakFields(importRecord, schemaBreak);
+
+			ModelValidationEngine.get().fireImportValidate(this, importRecord, schemaBreak, IImportInterceptor.TIMING_AFTER_IMPORT);
+			InterfaceWrapperHelper.save(schemaBreak);
+
+			importRecord.setM_DiscountSchemaBreak_ID(schemaBreak.getM_DiscountSchemaBreak_ID());
+		}
+
+		return schemaBreak;
+	}
+
+	private void setDiscountSchemaBreakFields(@NonNull final I_I_DiscountSchema importRecord, @NonNull final I_M_DiscountSchemaBreak schemaBreak)
```

Comment:
the logic about Pricing fields is not OK. 
it shall be:
* IsPriceOverride=N => PriceBase=n/a, Base_PricingSystem_ID=n/a, PriceStd=n/a, Std_AddAmt=n/a
* IsPriceOverride=Y, PriceBase=(P)ricingSystem => Base_PricingSystem_ID=set, PriceStd=n/a, Std_AddAmt=set
* IsPriceOverride=Y, PriceBase=(F)ixedPrice => Base_PricingSystem_ID=n/a PriceStd=set Std_AddAmt=n/a

btw, here is an example where I am setting all those price related fields: de.metas.ui.web.order.sales.pricingConditions.view.PricingConditionsRowsSaver.updatePrice(I_M_DiscountSchemaBreak, Price)